org.osgi.test.junit4/src/test/java/org/osgi/test/junit4/service/FilterArgumentsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) OSGi Alliance (2019). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.junit4.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.osgi.service.log.LogService;
+import org.osgi.test.common.annotation.InjectService;
+import org.osgi.test.common.service.BaseServiceUse;
+
+public class FilterArgumentsTest {
+
+	@Rule
+	public ServiceRule			sur	= new ServiceRule();
+
+	@InjectService(filter = "(%s=%s)", filterArguments = {
+		"objectClass", "org.osgi.service.log.LoggerFactory"
+	})
+	LogService					logService;
+	@InjectService(filter = "(%s=%s)", filterArguments = {
+		"objectClass", "org.osgi.service.log.LoggerFactory"
+	})
+	BaseServiceUse<LogService>	lsServiceAware;
```

Comment:
Shouldn't the field type be `ServiceAware<LogService>`?

Reply:
So provided the type is at least ServiceAware you can use it. :) there are a couple of our higher order types that have some added methods. I'll document those in another pass. For instance, this type let's me introspect the `ServiceConfiguration` details which I'm using the assert function.

Reply:
Using BaseServiceUse is implementation specific. You just happen to know it will work. But if I implemented ServiceAware or subclassed ServiceAware and tried to use that, the injection code would see type is assignable but fail to set the field. I am not sure using is assignable in the injection code is wise since we cannot support any subtype of ServiceAware (or List).

Reply:
fixed, now ckecks and expects exactly the types supported.